src/ds/d400/d400-factory.cpp

```diff
@@ -951,6 +951,227 @@ namespace librealsense
         }
     };
 
+    class rs436i_device : public d400_active,
+        public d400_color,
+        public d400_motion,
+        public ds_advanced_mode_base,
+        public firmware_logger_device
+    {
+    public:
+        rs436i_device(std::shared_ptr< const d400_info > const& dev_info, bool register_device_notifications)
+            : device(dev_info, register_device_notifications)
+            , backend_device(dev_info, register_device_notifications)
+            , d400_device(dev_info)
+            , d400_active(dev_info)
+            , d400_color(dev_info)
+            , d400_motion(dev_info)
+            , ds_advanced_mode_base(d400_device::_hw_monitor, get_depth_sensor())
+            , firmware_logger_device(
+                dev_info, d400_device::_hw_monitor, get_firmware_logs_command(), get_flash_logs_command())
+        {
+            check_and_restore_rgb_stream_extrinsic();
+            if (_fw_version >= firmware_version(5, 16, 0, 0))
```

Comment:
Don't think we need it the fw version check.
The FW that will support D436i is higher